### PR TITLE
feat(pit)

### PR DESCRIPTION
**Pull request by Kathund**
## Changes
Most of this class is from #358. There is now a players inv enderchest and amour. Theres a new class type that allows the for a pit item.  I have also added pit levels and prestige [(credit)](https://github.com/PitPanda/PitPandaProduction/blob/b1971f56ea1aa8c829b722cbb33247c96591c0cb/structures/Pit.js)

<details>
<summary>Screenshots</summary>

</details>

<details>
<summary>Checkboxes</summary>

- [x] I've added new features. (methods or parameters)
- [x] I've added jsdoc and typings.
- [ ] I've fixed bug. (_optional_ you can mention a issue if there is one)
- [ ] I've corrected the spelling in README, documentation, etc.
- [x] I've tested my code. (`npm run test`)
- [x] I've check for issues. (`npm run eslint`)
- [x] I've fixed my formatting. (`npm run prettier`)

</details>

### CODE REVIEW

File: src/structures/MiniGames/Pit.js [hunk 1]
```diff
@@ -8,12 +15,209 @@ class Pit {
...
+        const edited = [];
+        for (let i = 1; i < inventory.length; i++) {
+          if (!inventory[i].id) {
+            continue;
+          }
+          edited.push(new PitInventoryItem(inventory[i]));
+        }
+        return edited;
+      } catch (e) {
+        return e;
```
Comment (ChiefChippy2) [inline src/structures/MiniGames/Pit.js]:
```suggestion
        return [];
```

File: src/structures/MiniGames/Pit.js [hunk 2]
```diff
@@ -8,12 +15,209 @@ class Pit {
...
+        inventory = await decode(inventory.data);
+        const edited = [];
+        for (let i = 1; i < inventory.length; i++) {
+          if (!inventory[i].id) {
+            continue;
+          }
+          edited.push(new PitInventoryItem(inventory[i]));
+        }
+        return edited;
+      } catch (e) {
```
Comment (ChiefChippy2) [inline src/structures/MiniGames/Pit.js:170]:
```suggestion
      } catch {
```

File: src/structures/MiniGames/Pit.js [hunk 3]
```diff
@@ -8,12 +15,209 @@ class Pit {
...
+        chest = await decode(chest.data);
+        const edited = [];
+        for (let i = 1; i < chest.length; i++) {
+          if (!chest[i].id) {
+            continue;
+          }
+          edited.push(new PitInventoryItem(chest[i]));
+        }
+        return edited;
+      } catch (e) {
```
Comment (ChiefChippy2) [inline src/structures/MiniGames/Pit.js:192]:
```suggestion
      } catch {
```

File: src/structures/MiniGames/Pit.js [hunk 4]
```diff
@@ -8,12 +15,209 @@ class Pit {
...
+        return e;
```
Comment (ChiefChippy2) [inline src/structures/MiniGames/Pit.js]:
```suggestion
        return [];
```

File: src/structures/MiniGames/PitInventoryItem.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    /**
+     * Item name
+     * @type {string|null}
+     */
+    this.name = data?.tag?.display?.Name ? data.tag.display.Name.toString().replace(/§([1-9]|[a-f])|§/gm, '') : null;
+    /**
+     * Item lore
+     * @type {string|null}
+     */
+    this.lore = data?.tag?.display?.Lore ? data.tag.display.Lore.join('<br>') : null;
```
Comment (ChiefChippy2) [inline src/structures/MiniGames/PitInventoryItem.js]: my guy we are not in html
```suggestion
    this.lore = data?.tag?.display?.Lore ? data.tag.display.Lore.join('\n') : null;
```

**Review by ChiefChippy2 — CHANGES_REQUESTED** [review]
and also do the typings

File: src/structures/MiniGames/PitInventoryItem.js (hunk 5 above)
Comment (Kathund) [reply src/structures/MiniGames/PitInventoryItem.js]: I feel that it should stay like this. `\n` will spit it into new lines and make it look bad and its always been `<br>` in use. If this changes we should change it in [SkyblockInventoryItem](https://github.com/Hypixel-API-Reborn/hypixel-api-reborn/blob/8eedeb4e4d9d8081bbaf396f414f2d0731fb6ac7/src/structures/SkyBlock/SkyblockInventoryItem.js#L30) as well

Comment (ChiefChippy2) [reply src/structures/MiniGames/PitInventoryItem.js]: `<br>` is html standard, \n is text 
showing a non-programmer `<br>` would get them confused, where as a correctly rendered \n (newline) ... would appear as newline.

**Comment by Kathund** [comment]
@ChiefChippy2 waiting on review

**Review by ChiefChippy2 — APPROVED** [review]

**Pull request merged**